distributed/scheduler.py

```diff
@@ -4655,9 +4655,9 @@ def decide_worker(self, ts: TaskState):
         """
         Decide on a worker for task *ts*.  Return a WorkerState.
         """
-        valid_workers = self.valid_workers(ts)
+        valid_workers: set = self.valid_workers(ts)
 
-        if not valid_workers and not ts._loose_restrictions and self.workers:
+        if valid_workers == set() and not ts._loose_restrictions and self.workers:
```

Comment:
I'm surprised that it makes sense to create an empty set each time.

Reply:
Yeah I think you are right; we can do better. Will push a fix.

```python
In [4]: %timeit s == set()
91.3 ns ± 0.747 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: %timeit s is not None and not s
37.3 ns ± 0.373 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```